##### Before reading in data :

A bit of cleaning and tweeking was done with excel before reading the data into python:

On domains/subdomains of NCE, PR, and AE the following changes were made in excel: >99 to 99.1; >99.9 to 99.9; <1 to 0.1; <0.1 to 0.01; <0 to 0; <24 to 24 

On domains/subdomains of RDI the following changes were made: <1/90 to 1/90; >99/90 to 99/90
Also, errors to the RDI column in the form of dates were fixed to their intended fraction format

Additionally new columns were made: 
date tested column derived from one of the other test date columns which did have missing values, these were filled in with additional dates from other date columns (there should not be nulls)

one for age which shows years and months, or only days depending on the time difference between date of birth and date of testing

Three rows total associated with child IDs 523241 and 523644 were removed as there were no assemssments preformed accross all fields

##### Topic/Questions that need to be analyzed (MVP)

4. Does the child's age seem to impact their scoring?
6. Is there any difference in scoring noted based on evaluation type?

###### EDA and additional cleaning 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
import numpy as np 

In [2]:
df = pd.read_excel('../data/myversion_cleaneval.xlsx')

In [3]:
df.head()

Child ID Gender                    Age Date of Birth Date Tested  \
0    44879      F             8 month(s)    2022-09-27  2023-06-23   
1    47795      F  2 year(s)  2 month(s)    2020-12-23  2023-03-21   
2    54340      M  1 year(s)  9 month(s)    2021-06-22  2023-04-06   
3    54344      F            11 month(s)    2022-03-07  2023-02-23   
4    54515      M  1 year(s)  3 month(s)    2021-11-04  2023-02-24   

  Location - Sub Level 1                 Program Label  \
0              Southwest  BDI-3 Eligibility Evaluation   
1      Greater Nashville  BDI-3 Eligibility Evaluation   
2        First Tennessee  BDI-3 Eligibility Evaluation   
3    Southeast Tennessee  BDI-3 Eligibility Evaluation   
4          Memphis Delta  BDI-3 Eligibility Evaluation   

   Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
0                            7.0                             85.0   
1                           16.0                             88.0   
2                            8.0                             90.0   
3                           13.0                            115.0   
4                            5.0                             75.0   

   Adaptive Percentile Rank  ... Cognitive-Perception and Concepts RS  \
0                      16.0  ...                                  9.0   
1                      21.0  ...                                 11.0   
2                      25.0  ...                                 10.0   
3                      84.0  ...                                  9.0   
4                       5.0  ...                                 10.0   

  Cognitive-Perception and Concepts SS Cognitive-Perception and Concepts PR  \
0                                 13.0                                 84.0   
1                                  7.0                                 16.0   
2                                  7.0                                 16.0   
3                                 10.0                                 50.0   
4                                 10.0                                 50.0   

   Cognitive-Perception and Concepts AE  \
0                                  12.0   
1                                  18.0   
2                                  15.0   
3                                  12.0   
4                                  15.0   

   Cognitive-Perception and Concepts RDI  \
0                                  98/90   
1                                  29/90   
2                                  37/90   
3                                  92/90   
4                                  88/90   

   Cognitive-Perception and Concepts CSS  \
0                                  411.0   
1                                  437.0   
2                                  426.0   
3                                  411.0   
4                                  426.0   

   Cognitive-Perception and Concepts CSS 90%  \
0                                  392 - 431   
1                                  421 - 453   
2                                  408 - 444   
3                                  392 - 431   
4                                  408 - 444   

   Cognitive-Perception and Concepts Z-Score  \
0                                          1   
1                                     – 1.00   
2                                     – 1.00   
3                                          0   
4                                          0   

  Cognitive-Perception and Concepts T-Score  \
0                                      60.0   
1                                      40.0   
2                                      40.0   
3                                      50.0   
4                                      50.0   

  Cognitive-Perception and Concepts NCE  
0                                  71.0  
1                                  29.0  
2                                  29.0  
3                                  50.0  
4                                  50.0  



In [29]:
df.shape

(19498, 211)

In [30]:
df.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
Index: 19498 entries, 0 to 23943
Data columns (total 211 columns):
 #    Column                                                       Non-Null Count  Dtype         
---   ------                                                       --------------  -----         
 0    Child ID                                                     19498 non-null  object        
 1    Gender                                                       19498 non-null  object        
 2    Age                                                          19498 non-null  object        
 3    Date of Birth                                                19498 non-null  datetime64[ns]
 4    Date Tested                                                  19498 non-null  datetime64[ns]
 5    Location - Sub Level 1                                       19411 non-null  object        
 6    Program Label                                                19498 non-null  object        
 7    Adaptiv

In [31]:
df.columns

Index(['Child ID', 'Gender', 'Age', 'Date of Birth', 'Date Tested',
       'Location - Sub Level 1', 'Program Label',
       'Adaptive Sum of Scaled Scores', 'Adaptive Developmental Quotient',
       'Adaptive Percentile Rank',
       ...
       'Cognitive-Perception and Concepts RS',
       'Cognitive-Perception and Concepts SS',
       'Cognitive-Perception and Concepts PR',
       'Cognitive-Perception and Concepts AE',
       'Cognitive-Perception and Concepts RDI',
       'Cognitive-Perception and Concepts CSS',
       'Cognitive-Perception and Concepts CSS 90%',
       'Cognitive-Perception and Concepts Z-Score',
       'Cognitive-Perception and Concepts T-Score',
       'Cognitive-Perception and Concepts NCE'],
      dtype='object', length=211)

In [32]:
df=df.drop_duplicates().reset_index()

In [42]:
df=df.drop(columns=['index'])

After droping duplicate rows 19,498 out of 23,944 entries still remain

In [70]:
df.isnull().sum().sort_values(ascending=False)

Adaptive-Personal Responsibility RDI      9818
Motor-Perceptual Motor RDI                9813
Social Emotional-Peer Interaction RDI     9812
Motor-Perceptual Motor Date of Testing    9810
Motor-Perceptual Motor SS                 9810
                                          ... 
Program Label                                0
Date Tested                                  0
Date of Birth                                0
Age                                          0
Child ID                                     0
Length: 211, dtype: int64

In [71]:
#code from elisia
df.isna().sum().reset_index(name='nulls').sort_values('nulls').head(50)


index  nulls
0                                             Child ID      0
6                                        Program Label      0
4                                          Date Tested      0
55                         Adaptive-Self Care Examiner      0
2                                                  Age      0
3                                        Date of Birth      0
1                                               Gender      0
56                  Adaptive-Self Care Date of Testing      4
66                              Adaptive-Self Care NCE      4
65                          Adaptive-Self Care T-Score      4
64                          Adaptive-Self Care Z-Score      4
63                          Adaptive-Self Care CSS 90%      4
62                              Adaptive-Self Care CSS      4
59                               Adaptive-Self Care PR      4
60                               Adaptive-Self Care AE      4
58                               Adaptive-Self Care SS      4
57                               Adaptive-Self Care RS      4
13                                    Adaptive T-Score      5
12                                    Adaptive Z-Score      5
10                    Adaptive 95% Confidence Interval      5
9                             Adaptive Percentile Rank      5
8                      Adaptive Developmental Quotient      5
7                        Adaptive Sum of Scaled Scores      5
14                                        Adaptive NCE      5
82               Social Emotional-Adult Interaction SS      9
79         Social Emotional-Adult Interaction Examiner      9
80   Social Emotional-Adult Interaction Date of Tes...      9
81               Social Emotional-Adult Interaction RS      9
83               Social Emotional-Adult Interaction PR      9
89          Social Emotional-Adult Interaction T-Score      9
86              Social Emotional-Adult Interaction CSS      9
87          Social Emotional-Adult Interaction CSS 90%      9
88          Social Emotional-Adult Interaction Z-Score      9
90              Social Emotional-Adult Interaction NCE      9
84               Social Emotional-Adult Interaction AE      9
104  Social Emotional-Self Concept / Social Role Da...     12
106     Social Emotional-Self Concept / Social Role SS     12
107     Social Emotional-Self Concept / Social Role PR     12
108     Social Emotional-Self Concept / Social Role AE     12
110    Social Emotional-Self Concept / Social Role CSS     12
111  Social Emotional-Self Concept / Social Role CS...     12
112  Social Emotional-Self Concept / Social Role Z-...     12
113  Social Emotional-Self Concept / Social Role T-...     12
114    Social Emotional-Self Concept / Social Role NCE     12
103  Social Emotional-Self Concept / Social Role Ex...     12
105     Social Emotional-Self Concept / Social Role RS     12
125      Communication-Receptive Communication T-Score     13
124      Communication-Receptive Communication Z-Score     13
123     Communication-Receptive Communication CSS 90%      13
122          Communication-Receptive Communication CSS     13

In [43]:
df['Gender'].unique()

array(['F', 'M', 'U'], dtype=object)

In [44]:
df[df['Gender']=='U']
# 294

Child ID Gender                     Age Date of Birth Date Tested  \
149     486118      U   3 year(s)  1 month(s)    2020-03-30  2023-05-24   
193     487659      U  2 year(s)  11 month(s)    2020-07-29  2023-07-06   
213     488355      U  2 year(s)  11 month(s)    2020-08-24  2023-08-23   
221     488606      U  2 year(s)  10 month(s)    2020-07-23  2023-06-20   
233     489051      U              3 year(s)     2020-08-01  2023-08-09   
...        ...    ...                     ...           ...         ...   
19263  SP03057      U   2 year(s)  5 month(s)    2021-04-09  2023-09-27   
19276  SP03070      U   1 year(s)  9 month(s)    2021-12-18  2023-09-29   
19344  SP03138      U              4 month(s)    2023-05-23  2023-10-16   
19357  SP03151      U   2 year(s)  3 month(s)    2021-07-09  2023-10-19   
19360  SP03154      U   1 year(s)  4 month(s)    2022-05-31  2023-10-18   

      Location - Sub Level 1                 Program Label  \
149           East Tennessee       BDI-3 Annual Evaluation   
193           East Tennessee       BDI-3 Annual Evaluation   
213           East Tennessee       BDI-3 Annual Evaluation   
221         Upper Cumberland       BDI-3 Annual Evaluation   
233           East Tennessee       BDI-3 Annual Evaluation   
...                      ...                           ...   
19263         East Tennessee  BDI-3 Eligibility Evaluation   
19276         East Tennessee  BDI-3 Eligibility Evaluation   
19344        First Tennessee  BDI-3 Eligibility Evaluation   
19357         East Tennessee  BDI-3 Eligibility Evaluation   
19360        First Tennessee  BDI-3 Eligibility Evaluation   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
149                             21.0                            103.0   
193                             23.0                            109.0   
213                             18.0                             94.0   
221                             11.0                             73.0   
233                             21.0                            103.0   
...                              ...                              ...   
19263                           20.0                            100.0   
19276                            6.0                             80.0   
19344                            7.0                             85.0   
19357                            8.0                             64.0   
19360                            9.0                             95.0   

       Adaptive Percentile Rank  ... Cognitive-Perception and Concepts RS  \
149                        58.0  ...                                 21.0   
193                        73.0  ...                                 21.0   
213                        34.0  ...                                 23.0   
221                         4.0  ...                                  9.0   
233                        58.0  ...                                 14.0   
...                         ...  ...                                  ...   
19263                      50.0  ...                                 16.0   
19276                       9.0  ...                                 10.0   
19344                      16.0  ...                                  3.0   
19357                       1.0  ...                                 12.0   
19360                      37.0  ...                                  9.0   

      Cognitive-Perception and Concepts SS  \
149                                   10.0   
193                                   11.0   
213                                   12.0   
221                                    2.0   
233                                    7.0   
...                                    ...   
19263                                 11.0   
19276                                  7.0   
19344                                  6.0   
19357                                  7.0   
19360                                  7.0   

     

In [58]:
df[df['Gender']=='M']
# 11958

Child ID Gender                    Age Date of Birth Date Tested  \
2        54340      M  1 year(s)  9 month(s)    2021-06-22  2023-04-06   
4        54515      M  1 year(s)  3 month(s)    2021-11-04  2023-02-24   
5        54584      M  2 year(s)  8 month(s)    2020-06-05  2023-02-13   
12       55497      M  2 year(s)  5 month(s)    2021-04-29  2023-10-10   
14       55638      M  2 year(s)  8 month(s)    2020-11-05  2023-07-27   
...        ...    ...                    ...           ...         ...   
19489  SP03283      M  2 year(s)  7 month(s)    2021-04-13  2023-11-14   
19490  SP03284      M  2 year(s)  1 month(s)    2021-09-30  2023-11-14   
19491  SP03285      M  1 year(s)  7 month(s)    2022-04-07  2023-11-15   
19496  SP03290      M  3 year(s)  7 month(s)    2020-03-25  2023-10-26   
19497  SP03291      M             1 month(s)    2023-09-18  2023-10-24   

      Location - Sub Level 1                 Program Label  \
2            First Tennessee  BDI-3 Eligibility Evaluation   
4              Memphis Delta  BDI-3 Eligibility Evaluation   
5                  Northwest  BDI-3 Eligibility Evaluation   
12       Southeast Tennessee  BDI-3 Eligibility Evaluation   
14       Southeast Tennessee  BDI-3 Eligibility Evaluation   
...                      ...                           ...   
19489    Southeast Tennessee  BDI-3 Eligibility Evaluation   
19490        First Tennessee  BDI-3 Eligibility Evaluation   
19491        First Tennessee  BDI-3 Eligibility Evaluation   
19496          Memphis Delta       BDI-3 Annual Evaluation   
19497      Greater Nashville  BDI-3 Eligibility Evaluation   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
2                                8.0                             90.0   
4                                5.0                             75.0   
5                               17.0                             91.0   
12                              12.0                             76.0   
14                              16.0                             88.0   
...                              ...                              ...   
19489                            3.0                             49.0   
19490                            7.0                             61.0   
19491                           11.0                            105.0   
19496                           16.0                             88.0   
19497                           11.0                            105.0   

       Adaptive Percentile Rank  ... Cognitive-Perception and Concepts RS  \
2                         25.00  ...                                 10.0   
4                          5.00  ...                                 10.0   
5                         27.00  ...                                 12.0   
12                         5.00  ...                                  7.0   
14                        21.00  ...                                 12.0   
...                         ...  ...                                  ...   
19489                      0.01  ...                                 11.0   
19490                      1.00  ...                                 10.0   
19491                     63.00  ...                                  8.0   
19496                     21.00  ...                                 23.0   
19497                     63.00  ...                                  4.0   

      Cognitive-Perception and Concepts SS  \
2                                      7.0   
4                                     10.0   
5                                      6.0   
12                                     1.0   
14                                     6.0   
...                                    ...   
19489                                  5.0   
19490                                  6.0   
19491                                  5.0   
19496                                  9.0   
19497                                 12.0   

      Cognitive-P

In [59]:
df[df['Gender']=='F']
# 7246

Child ID Gender                    Age Date of Birth Date Tested  \
0        44879      F             8 month(s)    2022-09-27  2023-06-23   
1        47795      F  2 year(s)  2 month(s)    2020-12-23  2023-03-21   
3        54344      F            11 month(s)    2022-03-07  2023-02-23   
6        54861      F  2 year(s)  7 month(s)    2020-07-21  2023-03-17   
7        54937      F             1 year(s)     2022-03-09  2023-03-29   
...        ...    ...                    ...           ...         ...   
19487  SP03281      F             3 month(s)    2023-07-28  2023-11-14   
19492  SP03286      F  2 year(s)  2 month(s)    2021-08-16  2023-11-15   
19493  SP03287      F             1 month(s)    2023-09-25  2023-11-15   
19494  SP03288      F  2 year(s)  5 month(s)    2021-05-18  2023-11-13   
19495  SP03289      F             2 year(s)     2021-10-31  2023-11-14   

      Location - Sub Level 1                 Program Label  \
0                  Southwest  BDI-3 Eligibility Evaluation   
1          Greater Nashville  BDI-3 Eligibility Evaluation   
3        Southeast Tennessee  BDI-3 Eligibility Evaluation   
6              Memphis Delta  BDI-3 Eligibility Evaluation   
7        Southeast Tennessee  BDI-3 Eligibility Evaluation   
...                      ...                           ...   
19487          Memphis Delta  BDI-3 Eligibility Evaluation   
19492          Memphis Delta  BDI-3 Eligibility Evaluation   
19493          Memphis Delta  BDI-3 Eligibility Evaluation   
19494          South Central  BDI-3 Eligibility Evaluation   
19495          South Central  BDI-3 Eligibility Evaluation   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
0                                7.0                             85.0   
1                               16.0                             88.0   
3                               13.0                            115.0   
6                               16.0                             88.0   
7                               11.0                            105.0   
...                              ...                              ...   
19487                            9.0                             95.0   
19492                           30.0                            130.0   
19493                           13.0                            115.0   
19494                           25.0                            115.0   
19495                           21.0                            103.0   

       Adaptive Percentile Rank  ... Cognitive-Perception and Concepts RS  \
0                          16.0  ...                                  9.0   
1                          21.0  ...                                 11.0   
3                          84.0  ...                                  9.0   
6                          21.0  ...                                 18.0   
7                          63.0  ...                                 11.0   
...                         ...  ...                                  ...   
19487                      37.0  ...                                  6.0   
19492                      98.0  ...                                 17.0   
19493                      84.0  ...                                  6.0   
19494                      84.0  ...                                 20.0   
19495                      58.0  ...                                 13.0   

      Cognitive-Perception and Concepts SS  \
0                                     13.0   
1                                      7.0   
3                                     10.0   
6                                     11.0   
7                                     14.0   
...                                    ...   
19487                                 13.0   
19492                                 13.0   
19493                                 15.0   
19494                                 14.0   
19495                                 10.0   

      Cognitive-P

In [45]:
df['Child ID'].nunique()

19003

There are 19003 unique child id.s for this data set(19003 children)

In [46]:
df['Date of Birth'].min()
# oldest DOB 2019-10-15

Timestamp('2019-10-15 00:00:00')

In [47]:
df['Date of Birth'].max()
#youngest DOB 2023-10-30

Timestamp('2023-10-30 00:00:00')

In [48]:
df['Date Tested'].min()

Timestamp('2022-12-08 00:00:00')

In [49]:
df['Date Tested'].max()

Timestamp('2023-11-15 00:00:00')

looking at evaluations from December 8th 2022 - November 15th 2023//
11 months 7 days


In [87]:
df[18000:19355]
#when looking for duplicates there was SP****/t that showed up. this might have been a formating issue//waslooking for that here
#on a light over view, could not find what I thought I had seen previously

Child ID Gender                     Age Date of Birth Date Tested  \
18000  SP01794      M  2 year(s)  11 month(s)    2020-05-12  2023-05-01   
18001  SP01795      F   2 year(s)  8 month(s)    2020-08-18  2023-05-01   
18002  SP01796      M  2 year(s)  11 month(s)    2020-05-13  2023-05-01   
18003  SP01797      M  2 year(s)  11 month(s)    2020-05-22  2023-05-01   
18004  SP01798      F   1 year(s)  7 month(s)    2021-09-27  2023-05-01   
...        ...    ...                     ...           ...         ...   
19350  SP03144      F              3 month(s)    2023-07-07  2023-10-18   
19351  SP03145      M  1 year(s)  11 month(s)    2021-10-27  2023-10-17   
19352  SP03146      M   1 year(s)  5 month(s)    2022-05-11  2023-10-18   
19353  SP03147      M              2 year(s)     2021-09-27  2023-10-17   
19354  SP03148      F   1 year(s)  7 month(s)    2022-02-21  2023-10-18   

      Location - Sub Level 1                 Program Label  \
18000         East Tennessee       BDI-3 Annual Evaluation   
18001                    NaN       BDI-3 Annual Evaluation   
18002        First Tennessee       BDI-3 Annual Evaluation   
18003        First Tennessee       BDI-3 Annual Evaluation   
18004       Upper Cumberland       BDI-3 Annual Evaluation   
...                      ...                           ...   
19350         East Tennessee  BDI-3 Eligibility Evaluation   
19351          Memphis Delta  BDI-3 Eligibility Evaluation   
19352          Memphis Delta  BDI-3 Eligibility Evaluation   
19353          Memphis Delta  BDI-3 Eligibility Evaluation   
19354         East Tennessee  BDI-3 Eligibility Evaluation   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
18000                            3.0                             49.0   
18001                           17.0                             91.0   
18002                           14.0                             82.0   
18003                           14.0                             82.0   
18004                           13.0                            115.0   
...                              ...                              ...   
19350                            9.0                             95.0   
19351                            6.0                             80.0   
19352                            8.0                             90.0   
19353                           11.0                             73.0   
19354                            8.0                             90.0   

       Adaptive Percentile Rank  ... Cognitive-Perception and Concepts SS  \
18000                      0.01  ...                                  1.0   
18001                     27.00  ...                                  8.0   
18002                     12.00  ...                                  7.0   
18003                     12.00  ...                                  5.0   
18004                     84.00  ...                                 15.0   
...                         ...  ...                                  ...   
19350                     37.00  ...                                 12.0   
19351                      9.00  ...                                  1.0   
19352                     25.00  ...                                  5.0   
19353                      4.00  ...                                  3.0   
19354                     25.00  ...                                  8.0   

      Cognitive-Perception and Concepts PR  \
18000                                  0.1   
18001                                 25.0   
18002                                 16.0   
18003                                  5.0   
18004                                 95.0   
...                                    ...   
19350                                 75.0   
19351                                  0.1   
19352                                  5.0   
19353                                  1.0   
19354                                 25.0   

     

In [73]:
df['Location - Sub Level 1'].value_counts()

Location - Sub Level 1
Greater Nashville      3436
East Tennessee         3340
Memphis Delta          2715
South Central          2560
Upper Cumberland       2021
First Tennessee        1701
Southeast Tennessee    1583
Southwest              1100
Northwest               955
Name: count, dtype: int64

Greater Nashville has the most records
Northwest the least

In [74]:
df['Program Label'].value_counts()

Program Label
BDI-3 Eligibility Evaluation          12391
BDI-3 Annual Evaluation                6904
BDI-3 Milestone or Exit Evaluation      203
Name: count, dtype: int64

In [ ]:
More records of eligibility evals
Least records of exit evals

In [93]:
df.groupby(['Location - Sub Level 1','Program Label'])['Program Label'].count()
#program label per location 

Location - Sub Level 1  Program Label                     
East Tennessee          BDI-3 Annual Evaluation               1345
                        BDI-3 Eligibility Evaluation          1982
                        BDI-3 Milestone or Exit Evaluation      13
First Tennessee         BDI-3 Annual Evaluation                720
                        BDI-3 Eligibility Evaluation           965
                        BDI-3 Milestone or Exit Evaluation      16
Greater Nashville       BDI-3 Annual Evaluation               1039
                        BDI-3 Eligibility Evaluation          2392
                        BDI-3 Milestone or Exit Evaluation       5
Memphis Delta           BDI-3 Annual Evaluation                867
                        BDI-3 Eligibility Evaluation          1817
                        BDI-3 Milestone or Exit Evaluation      31
Northwest               BDI-3 Annual Evaluation                307
                        BDI-3 Eligibility Evaluation           612
   

In [79]:
df.groupby('Age')['Program Label'].count().sort_values(ascending=False)

Age
2 year(s)                 939
2 year(s)  1 month(s)     930
2 year(s)  10 month(s)    884
2 year(s)  11 month(s)    839
2 year(s)  6 month(s)     792
                         ... 
9 day(s)                    3
10 day(s)                   2
8 day(s)                    2
0 day(s)                    2
4 year(s)                   1
Name: Program Label, Length: 72, dtype: int64

In [61]:
df[df['Age']=='0 day(s)']
#only 2 records with child age as zero, might drop these, although bdi-3 doesn't have a minimum to test at

Child ID Gender       Age Date of Birth Date Tested  \
7648   547059      F  0 day(s)    2023-03-15  2023-03-15   
8640   549043      F  0 day(s)    2023-03-28  2023-03-28   

     Location - Sub Level 1                 Program Label  \
7648        First Tennessee  BDI-3 Eligibility Evaluation   
8640      Greater Nashville  BDI-3 Eligibility Evaluation   

      Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
7648                           11.0                            105.0   
8640                           12.0                            110.0   

      Adaptive Percentile Rank  ... Cognitive-Perception and Concepts RS  \
7648                      63.0  ...                                  3.0   
8640                      75.0  ...                                  5.0   

     Cognitive-Perception and Concepts SS  \
7648                                 10.0   
8640                                 14.0   

     Cognitive-Perception and Concepts PR  \
7648                                 50.0   
8640                                 91.0   

      Cognitive-Perception and Concepts AE  \
7648                                   0.0   
8640                                   5.0   

      Cognitive-Perception and Concepts RDI  \
7648                                  84/90   
8640                                  99/90   

      Cognitive-Perception and Concepts CSS  \
7648                                  331.0   
8640                                  370.0   

      Cognitive-Perception and Concepts CSS 90%  \
7648                                  311 - 350   
8640                                  355 - 386   

      Cognitive-Perception and Concepts Z-Score  \
7648                                          0   
8640                                       1.33   

     Cognitive-Perception and Concepts T-Score  \
7648                                      50.0   
8640                                      63.0   

     Cognitive-Perception and Concepts NCE  
7648                                  50.0  
8640                                  78.0  

[2 rows x 211 columns]

In [145]:
df['Age_num']=round(((df['Date Tested'] - df['Date of Birth']).dt.days) / 365,0)
#code from elisia
df['Age_num']=df['Age_num'].astype(int)

In [146]:
df.head(1)
#added another column for age.The column added in excel is lengthy and mixed object.
#redundant but might come in handy, if not, I'll drop it later

Child ID Gender         Age Date of Birth Date Tested  \
0    44879      F  8 month(s)    2022-09-27  2023-06-23   

  Location - Sub Level 1                 Program Label  \
0              Southwest  BDI-3 Eligibility Evaluation   

   Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
0                            7.0                             85.0   

   Adaptive Percentile Rank  ... Cognitive-Perception and Concepts SS  \
0                      16.0  ...                                 13.0   

  Cognitive-Perception and Concepts PR Cognitive-Perception and Concepts AE  \
0                                 84.0                                 12.0   

   Cognitive-Perception and Concepts RDI  \
0                                  98/90   

   Cognitive-Perception and Concepts CSS  \
0                                  411.0   

   Cognitive-Perception and Concepts CSS 90%  \
0                                  392 - 431   

   Cognitive-Perception and Concepts Z-Score  \
0                                          1   

   Cognitive-Perception and Concepts T-Score  \
0                                       60.0   

  Cognitive-Perception and Concepts NCE Age_num  
0                                  71.0       1  

[1 rows x 212 columns]

##### BDI evaluation portion (overall)

In [147]:
bdi_evals=df.iloc[:,[0,1,2,3,4,5,47,48,49,50,51,52,53,54,211]]
bdi_evals

Child ID Gender                    Age Date of Birth Date Tested  \
0        44879      F             8 month(s)    2022-09-27  2023-06-23   
1        47795      F  2 year(s)  2 month(s)    2020-12-23  2023-03-21   
2        54340      M  1 year(s)  9 month(s)    2021-06-22  2023-04-06   
3        54344      F            11 month(s)    2022-03-07  2023-02-23   
4        54515      M  1 year(s)  3 month(s)    2021-11-04  2023-02-24   
...        ...    ...                    ...           ...         ...   
19493  SP03287      F             1 month(s)    2023-09-25  2023-11-15   
19494  SP03288      F  2 year(s)  5 month(s)    2021-05-18  2023-11-13   
19495  SP03289      F             2 year(s)     2021-10-31  2023-11-14   
19496  SP03290      M  3 year(s)  7 month(s)    2020-03-25  2023-10-26   
19497  SP03291      M             1 month(s)    2023-09-18  2023-10-24   

      Location - Sub Level 1  BDI-3 Total Sum of Scaled Scores  \
0                  Southwest                              89.0   
1          Greater Nashville                             128.0   
2            First Tennessee                              77.0   
3        Southeast Tennessee                              91.0   
4              Memphis Delta                              56.0   
...                      ...                               ...   
19493          Memphis Delta                             112.0   
19494          South Central                             166.0   
19495          South Central                             118.0   
19496          Memphis Delta                             107.0   
19497      Greater Nashville                              89.0   

       BDI-3 Total Developmental Quotient  BDI-3 Total Percentile Rank  \
0                                    99.0                         47.0   
1                                    99.0                         47.0   
2                                    89.0                         23.0   
3                                   101.0                         53.0   
4                                    71.0                          3.0   
...                                   ...                          ...   
19493                               119.0                         90.0   
19494                               119.0                         90.0   
19495                                94.0                         34.0   
19496                                88.0                         21.0   
19497                                99.0                         47.0   

      BDI-3 Total 90% Confidence Interval BDI-3 Total 95% Confidence Interval  \
0                                  96-102                              95-103   
1                                  97-101                              96-102   
2                                   86-92                               85-93   
3                                  98-104                              97-105   
4                                   69-75                               68-76   
...                                   ...                                 ...   
19493                             116-122                             115-123   
19494                             117-121                             116-122   
19495                               92-96                               91-97   
19496                               86-90                               85-91   
19497                              96-102                              95-103   

      BDI-3 Total Z-Score  BDI-3 Total T-Score  BDI-3 Total NCE  Age_num  
0                  – 0.07                 49.0             49.0        1  
1                  – 0.07                 49.0             49.0        2  
2                  – 0.73                 43.0             35.0        2  
3                    0.07                 51.0             51.0        1  
4                  – 1.93                 31.0              9.0        1  
...         

In [148]:
bdi_evals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19498 entries, 0 to 19497
Data columns (total 15 columns):
 #   Column                               Non-Null Count  Dtype         
---  ------                               --------------  -----         
 0   Child ID                             19498 non-null  object        
 1   Gender                               19498 non-null  object        
 2   Age                                  19498 non-null  object        
 3   Date of Birth                        19498 non-null  datetime64[ns]
 4   Date Tested                          19498 non-null  datetime64[ns]
 5   Location - Sub Level 1               19411 non-null  object        
 6   BDI-3 Total Sum of Scaled Scores     19443 non-null  float64       
 7   BDI-3 Total Developmental Quotient   19443 non-null  float64       
 8   BDI-3 Total Percentile Rank          19443 non-null  float64       
 9   BDI-3 Total 90% Confidence Interval  19443 non-null  object        
 10  BDI-3 Tota

In [158]:
bdi_evals.isnull().sum()

Child ID                                 0
Gender                                   0
Age                                      0
Date of Birth                            0
Date Tested                              0
Location - Sub Level 1                  87
BDI-3 Total Sum of Scaled Scores        55
BDI-3 Total Developmental Quotient      55
BDI-3 Total Percentile Rank             55
BDI-3 Total 90% Confidence Interval     55
BDI-3 Total 95% Confidence Interval     55
BDI-3 Total Z-Score                    140
BDI-3 Total T-Score                    140
BDI-3 Total NCE                        140
Age_num                                  0
dtype: int64

In [174]:
bdi_evals_non_null['BDI-3 Total Developmental Quotient'].unique()

array([ 99.,  89., 101.,  71.,  94., 102.,  97.,  82.,  98.,  74.,  66.,
        88.,  69.,  86.,  79.,  40.,  92.,  95.,  85.,  70.,  61.,  44.,
        63.,  93.,  51.,  87.,  96.,  78.,  42.,  59.,  62.,  73.,  77.,
        72.,  52.,  84.,  75.,  90.,  55., 120., 123.,  47.,  65.,  64.,
        41.,  43., 106., 119., 100.,  48.,  68., 103.,  76.,  81., 133.,
        53.,  45., 115., 114.,  83., 107., 104., 116., 105.,  54.,  56.,
        67., 112., 113., 118., 108.,  46.,  91., 109.,  58., 160., 117.,
       126.,  57., 124., 122., 136.,  60.,  80., 111., 132.,  49., 110.,
       121.,  50., 131., 125., 128., 129., 130., 134., 127., 138., 142.,
       135., 143., 137., 141., 147., 152., 139.])

In [175]:
bdi_evals_non_null=bdi_evals.loc[~bdi_evals['BDI-3 Total Developmental Quotient'].isnull()]
#subsetting bdi-3 records that are not null in Developmental Quotient category

In [180]:
bdi_evals_non_null.info()

<class 'pandas.core.frame.DataFrame'>
Index: 19443 entries, 0 to 19497
Data columns (total 15 columns):
 #   Column                               Non-Null Count  Dtype         
---  ------                               --------------  -----         
 0   Child ID                             19443 non-null  object        
 1   Gender                               19443 non-null  object        
 2   Age                                  19443 non-null  object        
 3   Date of Birth                        19443 non-null  datetime64[ns]
 4   Date Tested                          19443 non-null  datetime64[ns]
 5   Location - Sub Level 1               19361 non-null  object        
 6   BDI-3 Total Sum of Scaled Scores     19443 non-null  float64       
 7   BDI-3 Total Developmental Quotient   19443 non-null  float64       
 8   BDI-3 Total Percentile Rank          19443 non-null  float64       
 9   BDI-3 Total 90% Confidence Interval  19443 non-null  object        
 10  BDI-3 Total 95%

In [179]:
bdi_evals_non_null.describe()

Date of Birth                    Date Tested  \
count                          19443                          19443   
mean   2021-08-25 23:08:04.925166080  2023-06-28 21:36:50.214473216   
min              2019-10-15 00:00:00            2022-12-08 00:00:00   
25%              2020-12-11 00:00:00            2023-04-21 00:00:00   
50%              2021-07-21 00:00:00            2023-07-03 00:00:00   
75%              2022-04-12 00:00:00            2023-09-10 12:00:00   
max              2023-10-30 00:00:00            2023-11-15 00:00:00   
std                              NaN                            NaN   

       BDI-3 Total Sum of Scaled Scores  BDI-3 Total Developmental Quotient  \
count                      19443.000000                        19443.000000   
mean                          89.863036                           87.687600   
min                            9.000000                           40.000000   
25%                           71.000000                           77.000000   
50%                           88.000000                           89.000000   
75%                          107.000000                           99.000000   
max                          199.000000                          160.000000   
std                           28.293704                           16.866901   

       BDI-3 Total Percentile Rank  BDI-3 Total T-Score  BDI-3 Total NCE  \
count                 19443.000000         19358.000000     19358.000000   
mean                     29.778581            41.925974        34.097293   
min                       0.010000            10.000000         0.100000   
25%                       6.000000            35.000000        18.000000   
50%                      23.000000            43.000000        35.000000   
75%                      47.000000            49.000000        49.000000   
max                      99.900000            90.000000        99.100000   
std                      26.415331            11.088515        21.183425   

            Age_num  
count  19443.000000  
mean       1.852389  
min        0.000000  
25%        1.000000  
50%        2.000000  
75%        3.000000  
max        4.000000  
std        0.982575

In [176]:
bdi_evals_non_null.loc[bdi_evals_non_null['BDI-3 Total Developmental Quotient']<79]

Child ID Gender                     Age Date of Birth Date Tested  \
4        54515      M   1 year(s)  3 month(s)    2021-11-04  2023-02-24   
10       55390      F   1 year(s)  8 month(s)    2021-11-01  2023-07-12   
12       55497      M   2 year(s)  5 month(s)    2021-04-29  2023-10-10   
14       55638      M   2 year(s)  8 month(s)    2020-11-05  2023-07-27   
19      404042      M  2 year(s)  11 month(s)    2020-03-17  2023-03-09   
...        ...    ...                     ...           ...         ...   
19475  SP03269      M   2 year(s)  5 month(s)    2021-05-12  2023-11-06   
19476  SP03270      M   2 year(s)  8 month(s)    2021-03-04  2023-11-08   
19484  SP03278      F   2 year(s)  6 month(s)    2021-04-17  2023-11-13   
19489  SP03283      M   2 year(s)  7 month(s)    2021-04-13  2023-11-14   
19490  SP03284      M   2 year(s)  1 month(s)    2021-09-30  2023-11-14   

      Location - Sub Level 1  BDI-3 Total Sum of Scaled Scores  \
4              Memphis Delta                              56.0   
10       Southeast Tennessee                              60.0   
12       Southeast Tennessee                              65.0   
14       Southeast Tennessee                              72.0   
19          Upper Cumberland                              15.0   
...                      ...                               ...   
19475        First Tennessee                              61.0   
19476         East Tennessee                              74.0   
19484        First Tennessee                              87.0   
19489    Southeast Tennessee                              65.0   
19490        First Tennessee                              78.0   

       BDI-3 Total Developmental Quotient  BDI-3 Total Percentile Rank  \
4                                    71.0                         3.00   
10                                   74.0                         4.00   
12                                   66.0                         1.00   
14                                   69.0                         2.00   
19                                   40.0                         0.01   
...                                   ...                          ...   
19475                                64.0                         1.00   
19476                                71.0                         3.00   
19484                                77.0                         6.00   
19489                                66.0                         1.00   
19490                                73.0                         4.00   

      BDI-3 Total 90% Confidence Interval BDI-3 Total 95% Confidence Interval  \
4                                   69-75                               68-76   
10                                  72-78                               71-79   
12                                  64-68                               63-69   
14                                  67-71                               66-72   
19                                  39-43                               38-44   
...                                   ...                                 ...   
19475                               62-66                               61-67   
19476                               69-73                               68-74   
19484                               75-79                               74-80   
19489                               64-68                               63-69   
19490                               71-75                               70-76   

      BDI-3 Total Z-Score  BDI-3 Total T-Score  BDI-3 Total NCE  Age_num  
4                  – 1.93                 31.0              9.0        1  
10                 – 1.73                 33.0             13.0        2  
12                 – 2.27                 27.0              2.0        2  
14                 – 2.07                 29.0              6.0        3  
19                    NaN                  NaN              NaN        3  
